darwin/dataset/remote_dataset.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import io
+import zipfile
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional
+
+import darwin
+from darwin.dataset.download_manager import download_all_images_from_annotations
+from darwin.dataset.upload_manager import _split_on_file_type, upload_file_to_s3
```

Comment:
Not sure how useful it is to split `download_manager` and `upload_manager` from here, since this is the only place we'll ever need those functions.

Reply:
I think reason would be code clarify and scalability, plus it makes the code easier to maintain (...adn the pythonic way of "explicit rather than implicit"); but I agree it is possible to merge these two files.